### PR TITLE
Add paddle deeponet model.

### PR DESCRIPTION
**Pull request by Liu-xiandong**
Add paddle deeponet model.

### CODE REVIEW

File: deepxde/nn/paddle/__init__.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 """Package for paddle NN modules."""
 
-__all__ = ["FNN"]
+__all__ = ["DeepONetCartesianProd", "FNN"]
 
 from .fnn import FNN
+from .deeponet import DeepONetCartesianProd
```
Comment (lululxvi) [inline deepxde/nn/paddle/__init__.py:6]:
```python
from .deeponet import DeepONetCartesianProd
from .fnn import FNN
```

File: deepxde/nn/paddle/deeponet.py [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+        x = paddle.einsum("bi,ni->bn", x_func, x_loc)
+        # Add bias
+        x += self.b
+
+        if self._output_transform is not None:
+            x = self._output_transform(inputs, x)
+        return x
+
+'''
+class DeepONet(NN):
```
Comment (lululxvi) [inline deepxde/nn/paddle/deeponet.py:70]: Delete these codes if they are not ready.

File: examples/operator/antiderivative_aligned.py [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
 # Load dataset
 d = np.load("antiderivative_aligned_train.npz", allow_pickle=True)
+# 2 train group (1000, 100)
 X_train = (d["X"][0].astype(np.float32), d["X"][1].astype(np.float32))
 y_train = d["y"].astype(np.float32)
 d = np.load("antiderivative_aligned_test.npz", allow_pickle=True)
 X_test = (d["X"][0].astype(np.float32), d["X"][1].astype(np.float32))
 y_test = d["y"].astype(np.float32)
 
+print("lxd_debug:")
```
Comment (lululxvi) [inline examples/operator/antiderivative_aligned.py:15]: Delete

File: examples/operator/antiderivative_aligned.py [hunk 4]
```diff
@@ -1,16 +1,20 @@
-"""Backend supported: tensorflow.compat.v1, tensorflow, pytorch"""
+"""Backend supported: tensorflow.compat.v1, tensorflow, pytorch, paddlepaddle"""
 import deepxde as dde
 import matplotlib.pyplot as plt
 import numpy as np
 
 # Load dataset
 d = np.load("antiderivative_aligned_train.npz", allow_pickle=True)
+# 2 train group (1000, 100)
```
Comment (lululxvi) [inline examples/operator/antiderivative_aligned.py:8]: What does this comment mean?

File: deepxde/model.py [hunk 5]
```diff
@@ -409,6 +409,11 @@ def outputs_losses(training, inputs, targets, losses_fn):
             else:
                 self.net.eval()
             inputs = paddle.to_tensor(inputs, stop_gradient=False)
+            if len(inputs) != 1:
+                inputs_tmp = []
+                for i in range(len(inputs)):
+                    inputs_tmp.append(paddle.to_tensor(inputs[i], stop_gradient=False))
+                inputs = inputs_tmp
```
Comment (lululxvi) [inline deepxde/model.py:416]:
```python
if isinstance(inputs, tuple):
    inputs = tuple(map(lambda x: paddle.to_tensor(x, stop_gradient=False), inputs))
else:
    inputs = paddle.to_tensor(inputs, stop_gradient=False)
```

File: deepxde/model.py [hunk 6]
```diff
@@ -409,6 +409,11 @@ def outputs_losses(training, inputs, targets, losses_fn):
             else:
                 self.net.eval()
             inputs = paddle.to_tensor(inputs, stop_gradient=False)
+            if len(inputs) != 1:
+                inputs_tmp = []
+                for i in range(len(inputs)):
+                    inputs_tmp.append(paddle.to_tensor(inputs[i], stop_gradient=False))
```
Comment (lululxvi) [inline deepxde/model.py:415]: You also need to modify line 404.

File: examples/operator/antiderivative_aligned.py [hunk 7]
```diff
@@ -11,6 +11,7 @@
 X_test = (d["X"][0].astype(np.float32), d["X"][1].astype(np.float32))
 y_test = d["y"].astype(np.float32)
 
+
```
Comment (lululxvi) [inline examples/operator/antiderivative_aligned.py:14]: Delete this empty line

Comment (Liu-xiandong) [reply examples/operator/antiderivative_aligned.py]: Done.

File: deepxde/model.py (hunk 5 above)
Comment (Liu-xiandong) [reply deepxde/model.py]: Done.

File: deepxde/model.py (hunk 6 above)
Comment (Liu-xiandong) [reply deepxde/model.py]: Done. Thanks~

**Comment by lululxvi** [comment]
Format your code via black https://github.com/psf/black

File: deepxde/model.py [hunk 8]
```diff
@@ -401,14 +401,23 @@ def outputs(training, inputs):
             else:
                 self.net.eval()
             with paddle.no_grad():
+                if isinstance(inputs, tuple):
+                    inputs = tuple(map(lambda x: paddle.to_tensor(x, stop_gradient=False), inputs))
+                else:
+                    inputs = paddle.to_tensor(inputs, stop_gradient=False)
```
[the remark this answers is not in the document]
Comment (Liu-xiandong) [reply deepxde/model.py:409]: Done.

**Comment by Liu-xiandong** [comment]
> Format your code via black https://github.com/psf/black

Done.

**Pull request merged**